db/db_impl/db_impl.cc

```diff
@@ -1503,6 +1503,9 @@ Status DBImpl::GetImpl(const ReadOptions& read_options,
       // However, we still want our callback to contain the actual snapshot so
       // that it can do the correct visibility filtering.
       callback->Refresh(snapshot);
+
+      // Internally, WriteUnpreparedTxnReadCallback::Refresh would set
+      // max_visible_seq = max(max_visible_seq, snapshot)
```

Comment:
What do you think about the comment that suggested we add an assert(callback->max_visible_seq() >= snapshot)?

Reply:
I forgot to submit my previous comments, but I tried it and it was hard, since only the wup callback does this.

Reply:
I cannot think of a case in WP where after Refresh this assert is not valid: assert(callback->max_visible_seq() >= snapshot)

Reply:
Actually, it seems to just be `InvalidSnapshotReadCallback` in wup recovery, so I can try to just fix the refresh override there.